### PR TITLE
Update documentation for TileLayer

### PR DESCRIPTION
**Pull request by kopp**
Document that getTileMetadata is not optional and what to return from it.

### CODE REVIEW

**Comment by kopp** [comment]
@felixpalmer : I think this was the solution to the problems I reported in #6816 .  Can you do me the honor of reviewing the pull request?

File: docs/api-reference/geo-layers/tile-layer.md [hunk 1]
```diff
@@ -314,11 +314,11 @@ Class that manages loading and purging of tile data. This class caches recently
 
 To implement a custom indexing scheme, extend `Tileset2D` and implement the following interface:
 
-- `getTileIndices({viewport, maxZoom, minZoom, zRange, modelMatrix, modelMatrixInverse})` - returns an array of indices in the given viewport. The indices should have the shape of Objects, like: `{q: '0123'}` to allow referencing in the URL template supplied to the `data` prop.
+- `getTileIndices({viewport, maxZoom, minZoom, zRange, modelMatrix, modelMatrixInverse})` - returns an array of indices in the given viewport. The indices should have the shape of Objects, like `{q: '0123'}`, to allow referencing in the URL template supplied to the `data` prop (`{q}` would get replaced with `'0123'` for index `{q: '0123'}`).
```
Comment (felixpalmer) [inline docs/api-reference/geo-layers/tile-layer.md:317]: I don't think this addition is needed

File: docs/api-reference/geo-layers/tile-layer.md [hunk 2]
```diff
@@ -314,11 +314,11 @@ Class that manages loading and purging of tile data. This class caches recently
...
 - `getTileId(index)` - returns unique string key for a tile index.
-- `getParentIndex(index)` - returns index of the parent tile.
+- `getParentIndex(index)` - returns the index of the parent tile.
 - `getTileZoom(index)` - returns a zoom level for a tile index.
-- `getTileMetadata(index) - returns additional metadata to add to tile (optional).
+- `getTileMetadata(index)` - returns additional metadata to add to tile.  This must be an object that contains a bounding box as `bbox` in the format `{west: number, south: number, east: number, north: number}`.
```
Comment (felixpalmer) [inline docs/api-reference/geo-layers/tile-layer.md:321]: Looks good 👍

**Review by felixpalmer — CHANGES_REQUESTED** [review]

File: docs/api-reference/geo-layers/tile-layer.md (hunk 1 above)
Comment (kopp) [reply docs/api-reference/geo-layers/tile-layer.md]: ok, removed :)

**Comment by kopp** [comment]
Thanks for looking at this.  I removed the first change.

**Review by felixpalmer — APPROVED** [review]

**Pull request merged**